manager/src/main/java/com/redhat/service/bridge/manager/BridgeWorker.java

```diff
@@ -0,0 +1,206 @@
+package com.redhat.service.bridge.manager;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.infra.models.dto.ManagedEntityStatus;
+import com.redhat.service.bridge.manager.dao.BridgeDAO;
+import com.redhat.service.bridge.manager.dao.PreparingWorkerDAO;
+import com.redhat.service.bridge.manager.models.Bridge;
+import com.redhat.service.bridge.manager.models.BridgeWorkerStatus;
+import com.redhat.service.bridge.manager.models.PreparingWorker;
+import com.redhat.service.bridge.manager.providers.InternalKafkaConfigurationProvider;
+import com.redhat.service.bridge.manager.providers.WorkerIdProvider;
+import com.redhat.service.bridge.rhoas.RhoasTopicAccessType;
+
+import io.quarkus.scheduler.Scheduled;
+import io.quarkus.vertx.ConsumeEvent;
+import io.vertx.mutiny.core.eventbus.EventBus;
+
+/**
+ * Accepted -> Preparing -> topic requested -> topic created -> provisioning
+ */
+
+@ApplicationScoped
+public class BridgeWorker implements AbstractWorker<Bridge> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BridgeWorker.class);
+
+    private static final String CREATE_KAFKA_TOPIC = "bridge-create-topic";
+    private static final String DELETE_KAFKA_TOPIC = "bridge-delete-topic";
+
+    @Inject
+    EventBus eventBus;
+
+    @Inject
+    BridgeDAO bridgeDAO;
+
+    @Inject
+    PreparingWorkerDAO preparingWorkerDAO;
+
+    @Inject
+    RhoasService rhoasService;
+
+    @Inject
+    InternalKafkaConfigurationProvider internalKafkaConfigurationProvider;
+
+    @Inject
+    WorkerIdProvider workerIdProvider;
+
+    @Override
+    public void accept(Bridge b) {
+        LOGGER.info("Accept entity " + b.getId());
+        if (b.getStatus() != null) {
+            LOGGER.warn("The status of the entity is not null. It will be overwritten.");
+        }
+        b.setStatus(ManagedEntityStatus.ACCEPTED);
```

Comment:
I think you may need to review the status's used.

See my comment below about the Shard processing `ACCEPTED` Bridges.

Reply:
See previous reply